File: editorial/chap08/cpp/answer_B54.cpp
#include <iostream>
#include <map>
using namespace std;
 
int N, A[100009];
map<int, int> Map;
 
int main() {
	// 입력
	cin >> N;
	for (int i = 1; i <= N; i++) cin >> A[i];
 
	// 답을 구한다
	long long Answer = 0;
	for (int i = 1; i <= N; i++) {
		Answer += Map[A[i]];
		Map[A[i]] += 1;
	}
 
	// 출력
	cout << Answer << endl;
	return 0;
}

File: editorial/chap08/python/answer_B54.py
Map = {}
 
# 입력
N = int(input())
A = [int(input()) for _ in range(N)]

# 답을 구한다
Answer = 0
for a in A:
	# dict.get에는 키가 존재하지 않는 경우의 기본값을 설정할 수 있다
	Answer += Map.get(a, 0)
	Map[a] = Map.get(a, 0) + 1

# 출력
print(Answer)
